src/navigation/planners/planners/node_ordered_mid_spline.py

```diff
@@ -120,23 +121,38 @@ def sort_cones(cones, start_index=None, end_index=None):
 class OrderedMapSpline(Node):
     spline_const = 10  # number of points per cone
     segment = int(spline_const * 0.1)  # percentage of PPC
+    planning = False
+    current_track = None
 
     def __init__(self):
         super().__init__("ordered_map_spline_node")
 
         # sub to track for all cone locations relative to car start point
         self.create_subscription(ConeDetectionStamped, "/slam/global_map", self.map_callback, 10)
+        self.create_subscription(UInt8, "/system/laps_completed", self.lap_callback, 10)
+        self.create_timer(0.1, self.planning_callback)
 
         # publishers
         self.qutms_path_pub: Publisher = self.create_publisher(QUTMSPathStamped, "/planner/path", 1)
         self.spline_path_pub: Publisher = self.create_publisher(Path, "/planner/spline_path", 1)
 
         self.get_logger().info("---Ordered path planner node initalised---")
 
+    def lap_callback(self, msg: UInt8):
+        if msg.data > 0:
+            self.planning = True
+            self.get_logger().info("Lap completed, planning commencing")
+
     def map_callback(self, track_msg: ConeDetectionStamped):
         self.get_logger().debug("Received map")
+        self.current_track = track_msg
+
+    def planning_callback(self):
+        # skip if we haven't completed a lap yet
+        if not self.planning:
+            return
 
-        cones = [cone_with_cov.cone for cone_with_cov in track_msg.cones_with_cov]
+        cones = self.current_track.cones
```

Comment:
Are you sure this doesn't need to be cone_with_cov's cone?

Reply:
the track actually populates both fields now, plain cones and cones with cov - so it's one less loop on the receiving end.